src/cps/search.cpp

```diff
@@ -397,6 +398,10 @@ namespace cps::search {
                 merge_result(comp.compile_flags, result.compile_flags);
                 merge_result(comp.link_libraries, result.link_libraries);
                 if (comp.type != loader::Type::interface) {
+                    if (!(comp.link_location || comp.location)) {
+                        return tl::make_unexpected(
+                            fmt::format("Component `{}` requires a 'link-location' or 'location' attribute", c_name));
+                    }
                     result.link_location.emplace_back(
                         prefix_replacer(comp.link_location.value_or(comp.location.value())));
```

Comment:
Without this change you get a segfault on this line from dereferencing `comp.location`.

Reply:
"location" is always required, but "link_location" is optional, so I think we should error if location is missing, regardless of whether link_location is set.

Reply:
Updated with that change, that makes sense... i was just trying to guard against the code on line 401 but if location is always required it makes sense to error in just that case.